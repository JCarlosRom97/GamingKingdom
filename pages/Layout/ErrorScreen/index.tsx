import React from "react";
import { useRouteError } from "react-router-dom";
export const ErrorScreen = () => {
  type errorProps = {
    statusText: string;
    message: string;
  };
  const error = useRouteError() as errorProps;
  return (
    <>
      <h1>OOPS!</h1>
      <p>This is an error screen</p>
      <p>Error: {error.statusText || error.message}</p>
    </>
  );
};
