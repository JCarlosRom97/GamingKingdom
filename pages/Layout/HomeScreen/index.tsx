import React, { useEffect } from "react";
import { HomeScreenLayout } from "./styles";
import { useSelector, useDispatch } from 'react-redux'
import { addNews } from "../../Redux/News/newsSlide";

export const HomeScreen = () => {

  const news = useSelector(({newsSlide})=> newsSlide)
  const dispatch = useDispatch()

  useEffect(()=>{
    dispatch(addNews({title:'Hola', description: 'Description'}))
    console.log(news) 
  },[])  

  return (
    <HomeScreenLayout>

    </HomeScreenLayout>
  );
};
