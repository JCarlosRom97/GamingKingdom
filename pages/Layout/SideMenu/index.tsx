import React from "react";
import {
  SideBar,
  Title,
  SideBarSearch,
  Button,
  MenuItem,
  SideBarSearchContainer,
  SideBarContent,
  Content,   
  LayoutWrapper,
  LinkMenu
} from "./styles";

import { usePathname } from "next/navigation";

type SideMenuProps = {
  children?: React.ReactElement
}

export default function SideMenu({children}: SideMenuProps) {
  const pathname = usePathname()

  return (
    <LayoutWrapper>
      {/* Side Bar */}
      <SideBar id="sidebar">
        <Title >Gaming Kingdom</Title>
        <SideBarSearchContainer>
          <SideBarSearch
            id="q"
            aria-label="Search contacts"
            placeholder="Search"
            type="search"
            name="q"
          />
          <Button>Go</Button>
        </SideBarSearchContainer>
        <SideBarContent>
          <MenuItem >
            <LinkMenu isactive={pathname === '/' ? 1 : 0} href={'/'}>Home</LinkMenu>
          </MenuItem>
          <MenuItem >
            <LinkMenu isactive={pathname === '/dashboard' ? 1 : 0} href={'/dashboard'}>Dashboard</LinkMenu>
          </MenuItem>
        </SideBarContent>
      </SideBar>
      {/* Content of page */}
      <Content>
        {children}
      </Content>
    </LayoutWrapper>
  );
}
