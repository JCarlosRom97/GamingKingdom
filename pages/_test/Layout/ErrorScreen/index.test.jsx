import { render } from "@testing-library/react";
import { ErrorScreen } from "../../../Layout/ErrorScreen";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useRouteError: () => jest.fn(),
}));

test("renders correctly", () => {
  const { container } = render(
    <BrowserRouter>
      <ErrorScreen/>
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();
});
