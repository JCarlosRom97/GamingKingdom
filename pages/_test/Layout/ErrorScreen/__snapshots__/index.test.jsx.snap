// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    OOPS!
  </h1>
  <p>
    This is an error screen
  </p>
  <p>
    Error: 
  </p>
</div>
`;
